plugins/modules/omevv_baseline_profile.py

```diff
@@ -0,0 +1,589 @@
+from ansible_collections.dellemc.openmanage.plugins.module_utils.utils import validate_job_wait, validate_time
+from ansible_collections.dellemc.openmanage.plugins.module_utils.omevv_utils.omevv_firmware_utils import OMEVVFirmwareProfile, OMEVVBaselineProfile
+from ansible_collections.dellemc.openmanage.plugins.module_utils.omevv_utils.omevv_info_utils import OMEVVInfo
+
+
+SUCCESS_CREATION_MSG = "Successfully created the baseline profile."
+FAILED_CREATION_MSG = "Unable to create the baseline profile."
+SUCCESS_MODIFY_MSG = "Successfully modified the baseline profile."
+FAILED_MODIFY_MSG = "Unable to modify the baseline profile."
+SUCCESS_DELETION_MSG = "Successfully deleted the baseline profile."
+FAILED_DELETION_MSG = "Unable to delete the baseline profile."
+PROFILE_NOT_FOUND_MSG = "Unable to delete the profile {profile_name} because the profile name is invalid. Enter a valid profile name and retry the operation."
+CHANGES_FOUND_MSG = "Changes found to be applied."
+CHANGES_NOT_FOUND_MSG = "No changes found to be applied."
+TIMEOUT_NEGATIVE_OR_ZERO_MSG = "The value for the 'job_wait_timeout' parameter cannot be negative or zero."
+UNREACHABLE_MSG = "The URL with the {ip}:{port} cannot be reached."
+SOURCE_NOT_FOUND_MSG = "The Requested resource cannot be found."
+
+
+class BaselineProfile:
+
+    def __init__(self, module, rest_obj):
+        self.module = module
+        self.obj = rest_obj
+        self.omevv_baseline_obj = OMEVVBaselineProfile(self.obj)
+        self.omevv_profile_obj = OMEVVFirmwareProfile(self.obj)
+        self.omevv_info_obj = OMEVVInfo(self.obj)
+
+    def validate_common_params(self):
+        resp = validate_job_wait(self.module)
+        if resp:
+            self.module.exit_json(msg=TIMEOUT_NEGATIVE_OR_ZERO_MSG, failed=True)
+        validate_time(self.module.params.get('time'), self.module)
+
+        repository_profile = self.module.params.get('repository_profile')
+        repo_validation = self.omevv_baseline_obj.validate_repository_profile(repository_profile)
+        if "error" in repo_validation:
+            self.module.exit_json(msg=repo_validation["error"], failed=True)
+
+        cluster_names = self.module.params.get('cluster')
+        vcenter_uuid = self.module.params.get('vcenter_uuid')
+        cluster_validation = self.omevv_baseline_obj.validate_cluster_names(cluster_names, vcenter_uuid)
+        if "error" in cluster_validation:
+            self.module.exit_json(msg=cluster_validation["error"], failed=True)
+
+    def execute(self):
+        """To be overridden by subclasses to implement specific profile creation or deletion logic."""
+        pass
+
+
+class CreateBaselineProfile(BaselineProfile):
+
+    def __init__(self, module, rest_obj, existing_profile=None):
+        super().__init__(module, rest_obj)
+        self.existing_profile = existing_profile
+
+    def diff_mode_check(self, payload):
```

Comment:
any reason you are not using recursive_diff function and doing it yourself, it may lead to issue. Just check if you can utilize the function

Reply:
tested it implemented my own logic